app/src/main/java/com/kamron/pogoiv/OcrHelper.java

```diff
@@ -148,28 +148,77 @@ private double getPokemonLevelFromImg(Bitmap pokemonImage, int trainerLevel) {
      * If there was no detected upgrade cost, returns -1
      *
      * @param pokemonImage The image of the full pokemon screen
-     * @return the evolution cost, or -1 on scan failure
+     * @return the evolution cost, or -1 on no upgrade cost, or -999 on scan failure
      */
     public int getPokemonEvolutionCostFromImg(Bitmap pokemonImage) {
+        int error = -999;
```

Comment:
Exceptions are usually recommended for exceptional errors, not for all of them.

The rest of this code is now using Guava's `Optional` instead of special defaults—it's impossible to use `Optional.absent` as an integer 😉 . (The only caveat is that you should never use `.get`without checking `isPresent`). While that's not as good as a proper `Optional` from  Java 8, I'll take what I can get ;-).